### ISSUE TITLE
[unity] Error reading skeleton JSON file for SkeletonData asset: New SkeletonDataAsset Specified cast is not valid.

### ISSUE THREAD

**Issue by TowerCrow** [EsotericSoftware/spine-runtimes#1368]
Issues with the latest 3.8 beta and Unity runtimes. Unity does not automatically generate a skeleton Data Asset and I can't create it manually either because I get the following errors:

```
Error reading skeleton JSON file for SkeletonData asset: New SkeletonDataAsset
Specified cast is not valid.
  at Spine.SkeletonJson.ReadSkeletonData (System.IO.TextReader reader) [0x00bf4] in C:\GameDev\Spooky Starlets\Assets\Spine\Runtime\spine-csharp\SkeletonJson.cs:262 
  at Spine.Unity.SkeletonDataAsset.ReadSkeletonData (System.String text, Spine.AttachmentLoader attachmentLoader, System.Single scale) [0x00017] in C:\GameDev\Spooky Starlets\Assets\Spine\Runtime\spine-unity\Asset Types\SkeletonDataAsset.cs:234 
  at Spine.Unity.SkeletonDataAsset.GetSkeletonData (System.Boolean quiet) [0x000c1] in C:\GameDev\Spooky Starlets\Assets\Spine\Runtime\spine-unity\Asset Types\SkeletonDataAsset.cs:170 
UnityEngine.Debug:LogError(Object, Object)
Spine.Unity.SkeletonDataAsset:GetSkeletonData(Boolean) (at Assets/Spine/Runtime/spine-unity/Asset Types/SkeletonDataAsset.cs:174)
Spine.Unity.Editor.SkeletonInspectorPreview:Initialize(Action, SkeletonDataAsset, String) (at Assets/Spine/Editor/spine-unity/Editor/SkeletonDataAssetInspector.cs:784)
Spine.Unity.Editor.SkeletonDataAssetInspector:OnInspectorGUI() (at Assets/Spine/Editor/spine-unity/Editor/SkeletonDataAssetInspector.cs:203)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
```

```
Error reading skeleton JSON file for SkeletonData asset: New SkeletonDataAsset
Specified cast is not valid.
  at Spine.SkeletonJson.ReadSkeletonData (System.IO.TextReader reader) [0x00bf4] in C:\GameDev\Spooky Starlets\Assets\Spine\Runtime\spine-csharp\SkeletonJson.cs:262 
  at Spine.Unity.SkeletonDataAsset.ReadSkeletonData (System.String text, Spine.AttachmentLoader attachmentLoader, System.Single scale) [0x00017] in C:\GameDev\Spooky Starlets\Assets\Spine\Runtime\spine-unity\Asset Types\SkeletonDataAsset.cs:234 
  at Spine.Unity.SkeletonDataAsset.GetSkeletonData (System.Boolean quiet) [0x000c1] in C:\GameDev\Spooky Starlets\Assets\Spine\Runtime\spine-unity\Asset Types\SkeletonDataAsset.cs:170 
UnityEngine.Debug:LogError(Object, Object)
Spine.Unity.SkeletonDataAsset:GetSkeletonData(Boolean) (at Assets/Spine/Runtime/spine-unity/Asset Types/SkeletonDataAsset.cs:174)
Spine.Unity.Editor.SkeletonInspectorPreview:Initialize(Action, SkeletonDataAsset, String) (at Assets/Spine/Editor/spine-unity/Editor/SkeletonDataAssetInspector.cs:784)
Spine.Unity.Editor.SkeletonDataAssetInspector:OnInteractivePreviewGUI(Rect, GUIStyle) (at Assets/Spine/Editor/spine-unity/Editor/SkeletonDataAssetInspector.cs:682)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)```

**Comment by TowerCrow** [comment]
It also shows this
```
Skeleton 'girl' (exported with Spine 3.8.12-beta) may be incompatible with your runtime version: spine-csharp v3.7
UnityEngine.Debug:LogWarningFormat(String, Object[])
Spine.Unity.Editor.AssetUtility:IsSpineData(TextAsset) (at Assets/Spine/Editor/spine-unity/Editor/SpineEditorUtilities.cs:1271)
Spine.Unity.Editor.AssetUtility:ImportSpineContent(String[], Boolean) (at Assets/Spine/Editor/spine-unity/Editor/SpineEditorUtilities.cs:847)
Spine.Unity.Editor.AssetUtility:HandleOnPostprocessAllAssets(String[]) (at Assets/Spine/Editor/spine-unity/Editor/SpineEditorUtilities.cs:673)
Spine.Unity.Editor.SpineEditorUtilities:OnPostprocessAllAssets(String[], String[], String[], String[]) (at Assets/Spine/Editor/spine-unity/Editor/SpineEditorUtilities.cs:218)
UnityEditor.AssetPostprocessingInternal:PostprocessAllAssets(String[], String[], String[], String[], String[])
UnityEditorInternal.InternalEditorUtility:ProjectWindowDrag(HierarchyProperty, Boolean)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

```
But I'm sure I've imported the 3.8 package.

**Comment by NathanSweet** [comment]
I replied on your forum thread here:
http://esotericsoftware.com/forum/Error-reading-skeleton-JSON-file-for-SkeletonData-asset-12051

We should definitely fix the error message you got saying your runtime version is `v3.7`!

**Comment by NathanSweet** [comment]
Note you'll need to use the [skin-bones branch](https://github.com/EsotericSoftware/spine-runtimes/tree/skin-bones) until it is merged into 3.8-beta (soon).
